app.rb

```diff
@@ -0,0 +1,138 @@
+require './person'
+require './student'
+require './rental'
+require './classroom'
+require './teacher'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def home
+    puts 'Welcome to School Library Application'
+    until options_list
+      input = gets.chomp.to_i
+      if input == 7
+        puts 'Thanks for using School Library...'
+        exit
+      end
+
+      option input
+    end
+  end
+
+  def list_persons
+    puts 'Person list is empty. Please add a person to your list.' if @persons.empty?
+    @persons.each do |person|
+      puts "Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def list_books
+    puts 'List of Books is empty. Please add a Book to your ist.' if @books.empty?
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def add_person
+    puts 'Who do you want to add?'
+    puts '1. Student'
+    puts '2. Teacher'
+    input = gets.chomp.to_i
+    if input == 1
+      add_student
+    elsif input == 2
+      add_teacher
+    else
+      puts 'Invalid Option Input. Retry'
```

Comment:
- [ ] When I try adding a person an error is thrown as seen below. Kindly improve upon the same. 

![image](https://github.com/Cosmosha/school_library/assets/12745474/43b5f6e2-50a5-4be9-8f3d-02b23f2b9c55)